pgmax/fg/graph.py

```diff
@@ -736,8 +744,12 @@ def __setitem__(
         object.__setattr__(
             self,
             "value",
-            update_evidence(
-                jax.device_put(self.value), {name: jax.device_put(data)}, self.fg_state
+            np.asarray(
+                update_evidence(
+                    jax.device_put(self.value),
+                    {name: jax.device_put(data)},
+                    self.fg_state,
+                ),
             ),
```

Comment:
Not expert on jax but for my own clarification. 
Do we not need to clear the variables `self.value` and `data` from jax memory memory?

Reply:
`self.value` and `data` would stay on CPU. Intermediate objects on GPU will be cleared automatically if you use the `platform` option. 

The leak came from a failure to clear large jit compilation cache (the cache is large because it involves a large constant array, i.e. the wiring). A completely leak-free way is to include wiring as part of the arguments, but in my experiments that leads to slow compiling and hurts performance. Current solution is a bit of a compromise but should suffice for most cases.